### PR TITLE
Allow a user to choose if nginx is deployed as part of installation

### PR DESCRIPTION
**Pull request by mark-au**
A user may want to deploy nginx-ingress as part of their own installation process. Support a boolean var of `deploy_nginx_software` which defaults to true and when negative create a OCI loadbalancer but skip the installation of nginx-ingress to Kubernetes

Tested with negative and positive value.
Signed-off-by: Mark Cram <mark.cram@oracle.com>

### CODE REVIEW

**Comment by Djelibeybi** [comment]
Could you please replace any user-visible string to `NGINX`as it's a registered trademark.

**Comment by mark-au** [comment]
> Could you please replace any user-visible string to `NGINX`as it's a registered trademark.

Will do, I internally debated this as they use `nginx` in their documentation and `Nginx` in their licensing

**Review by Djelibeybi — APPROVED** [review]

**Comment by Djelibeybi** [comment]
I meant to link to this earlier: https://www.bizapedia.com/trademarks/nginx-87487184.html which shows the "mark identifier" as "NGINX".

**Comment by hyder** [comment]
If the user doesn't install an Ingress Controller and set the ServiceType to NodePort as we are doing here, they won't be able to access their services. So I'm wondering what is the purpose of giving the choice to turn it off.

**Comment by mark-au** [comment]
> If the user doesn't install an Ingress Controller and set the ServiceType to NodePort as we are doing here, they won't be able to access their services. So I'm wondering what is the purpose of giving the choice to turn it off.

Other projects might perform their own nginx-ingress installation, I understand your default is to install this non-OLCNE component but it should be a user option.

File: modules/olcne/variables.tf [hunk 1]
```diff
@@ -100,3 +100,9 @@ variable "container_registry_urls" {
   }
   type = map(string)
 }
+
+variable "deploy_nginx_software" {
```
Comment (hyder) [inline modules/olcne/variables.tf:104]: no need for description and default here. You've already set them in root variables.

Comment (hyder) [reply modules/olcne/variables.tf:104]: > > If the user doesn't install an Ingress Controller and set the ServiceType to NodePort as we are doing here, they won't be able to access their services. So I'm wondering what is the purpose of giving the choice to turn it off.
> 
> Other projects might perform their own nginx-ingress installation, I understand your default is to install this non-OLCNE component but it should be a user option.

Fair enough and you've left the default to true 👍 I'll merge after you make the changes.

Comment (mark-au) [reply modules/olcne/variables.tf:104]: > no need for description and default here. You've already set them in root variables.

Removed

**Pull request merged**